### PR TITLE
Comments and cleanup

### PR DESCRIPTION
**Pull request by kubijo**
Hi.

First and foremost thank you for work you have done on this. I'm coming from an app where we use a lot of tooling around React and Redux, but a lot of this was black magic to me (hot reloading and some test cases especially).

That said, to understand it I've gone through the project and commented non-obvious parts around the build setup with quick description and links to further details + fixed few code style bits + removed tools dir where there was only a stale submodule.

This is first of all helpful to me and potentially for my team, however it may be useful to other newcomers as well.

If you'll find it useful, but find some glitches or things that are not true, I'm ready to clean it up.

### CODE REVIEW

File: webpack.config.base.js [hunk 1]
```diff
@@ -11,18 +15,23 @@ export default {
       loader: 'json-loader'
     }]
   },
+
   output: {
     path: path.join(__dirname, 'dist'),
     filename: 'bundle.js',
+
+    // https://github.com/webpack/webpack/issues/1114
```
Comment (pushtheworldllc) [inline webpack.config.base.js]: This link might have just helped me a bunch! Thanks!

File: webpack.config.electron.js [hunk 2]
```diff
@@ -8,31 +12,47 @@ export default {
...
+    // Add source map support for stack traces in node
+    // https://github.com/evanw/node-source-map-support
     new webpack.BannerPlugin(
       'require("source-map-support").install();',
       { raw: true, entryOnly: false }
     ),
+
+    // NODE_ENV should be production so that modules do not perform certain development checks
     new webpack.DefinePlugin({
       'process.env': {
```
Comment (pushtheworldllc) [inline webpack.config.electron.js]: Should this also be `process.env.NODE_ENV` like in `webpack.config.development.js`?

Comment (pushtheworldllc) [inline webpack.config.electron.js]: Good time to make this match `webpack.config.development.js`

`'process.env.NODE_ENV': JSON.stringify('production')`

**Comment by pushtheworldllc** [comment]
Awesome job @kubijo! Did you guys build with any native modules? Trying to figure out the right way to build them, I've been failing for about 8 months now

File: webpack.config.electron.js [hunk 3]
```diff
@@ -8,31 +12,47 @@ export default {
...
         NODE_ENV: JSON.stringify('production')
       }
     })
   ],
 
+  /**
+   * Set targed to Electron speciffic node.js env.
+   * https://github.com/chentsulin/webpack-target-electron-renderer#how-this-module-works
```
Comment (pushtheworldllc) [inline webpack.config.electron.js]: This link brings you to describing how the render web pack works not the main as expected.

**Comment by amilajack** [comment]
@aj-ptw which part is failing?

**Comment by amilajack** [comment]
@kubijo thanks for the PR! Can you please rebase this and fix merge conflicts?

**Comment by kubijo** [comment]
I've got to it and rebased over master.
Can you however please check yourself that those changes are still ok and relevant?

These build configs have changed a bit since I've seen them the last time so it would take me a lot to get the context again.

Particularly `webpack.config.test.js:14` seems to be out of date (wrong assumption, since babel loader is no longer present), but I won't go into that in this PR.

File: webpack.config.test.js [hunk 4]
```diff
@@ -1,13 +1,16 @@
...
+const devConfig = require('./webpack.config.development');
 
 module.exports = validate({
   output: {
     libraryTarget: 'commonjs2'
   },
   module: {
-    loaders: devConfigs.module.loaders.slice(1)  // remove babel-loader
+    // Use base + development loaders, but exclude 'babel-loader'
+    loaders: devConfig.module.loaders.slice(1)
```
Comment (kubijo) [inline webpack.config.test.js]: It seems to me that `devConfig.module.loaders` does not contain babel-loader … did that change or do I not see something?

**Comment by kubijo** [comment]
@amilajack I see, that's one thing I was confused about. Will remove.

**Comment by amilajack** [comment]
Merged :party

**Pull request merged**